core/src/java/show/ezkz/casprzak/core/tools/functiongenerators/TaylorSeries.java

```diff
@@ -0,0 +1,120 @@
+package show.ezkz.casprzak.core.tools.functiongenerators;
+
+import show.ezkz.casprzak.core.config.Settings;
+import show.ezkz.casprzak.core.functions.GeneralFunction;
+import show.ezkz.casprzak.core.functions.binary.Pow;
+import show.ezkz.casprzak.core.functions.commutative.Product;
+import show.ezkz.casprzak.core.functions.commutative.Sum;
+import show.ezkz.casprzak.core.functions.endpoint.Constant;
+import show.ezkz.casprzak.core.functions.endpoint.Variable;
+import show.ezkz.casprzak.core.tools.MiscTools;
+import show.ezkz.casprzak.core.tools.VariableTools;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The methods in {@link TaylorSeries} produce Taylor Series for functions.
+ */
+public class TaylorSeries {
+
+    private TaylorSeries(){}
+
+    /**
+     * Returns a Maclaurin series of the specified function of a given degree
+     * @param function the function whose Maclaurin series is being found
+     * @param degree the degree of the Maclaurin polynomial
+     * @return a Maclaurin series of the specified function
+     */
+    public static GeneralFunction makeTaylorSeries(GeneralFunction function, int degree) {
+        return makeTaylorSeries(function, degree, 0);
```

Comment:
The control flow would probably make more sense to call the below method rather than `makeTaylorSeries(Function, int, double)`